test/runtime/unlink.c

```diff
@@ -45,6 +48,53 @@ static void test_unlinkat(int dirfd)
     }
 }
 
+static void test_tmpfile()
+{
+    int fd;
+    struct statfs statbuf;
+    char buf[8192];
+    char vbuf[8192];
+
+    if (mkdir("/tmp", 0777) != 0 && errno != EEXIST) {
+        perror("mkdir /tmp");
+        exit(EXIT_FAILURE);
+    }
+    statfs("/", &statbuf);
+    unsigned long fsfree = statbuf.f_bfree;
+    fd = open("/tmp", O_RDONLY|O_TMPFILE);
+    if (fd != -1) {
+        perror("tmpfile rdonly open");
+        exit(EXIT_FAILURE);
+    }
+    fd = open("/tmp", O_RDWR|O_TMPFILE);
+    if (fd < 0) {
+        perror("tmpfile open");
+        exit(EXIT_FAILURE);
+    }
+    for (int i = 0; i < sizeof(buf); i++)
+        buf[i] = i % sizeof(buf[0]);
```

Comment:
sizeof(buf[0]) is 1, so this is basically setting buf[i] to a constant value (0); did you mean something else?

Reply:
No, it just needs to be a straight assignment. I've fixed it.